dependabot/update_version_data.py

```diff
@@ -0,0 +1,298 @@
+    module_details_json = get_immediate_dependents(module_name_list, module_details_json)
+    print('Fetched immediate dependents of each module')
+    module_details_json = calculate_levels(module_name_list, module_details_json)
+    print('Generated module dependency graph and updated module levels')
+    module_details_json['modules'].sort(key=lambda s: s['level'])
+    update_json_file(module_details_json)
+    print('Updated module details successfully')
+    commit_json_file()
+    print("Updated module details in 'ballerina-release' successfully")
+
+# Sorts the ballerina extension module list in ascending order
+def sort_module_name_list():
+    try:
+        with open(MODULE_LIST_FILE) as f:
+            name_list = json.load(f)
+    except:
+        print('Failed to read module_list.json')
+        sys.exit()
+
+    name_list['modules'].sort(key=lambda x: x.split('-')[-1])
+    
+    try:
+        with open(MODULE_LIST_FILE, 'w') as json_file:
+            json_file.seek(0) 
+            json.dump(name_list, json_file, indent=4)
+            json_file.truncate()
+    except:
+        print('Failed to write to file module_list.json')
+        sys.exit()
+        
+    return name_list['modules'] 
+
+# Returns the file in the given url
+# Retry decorator will retry the function 3 times, doubling the backoff delay if URLError is raised 
+@retry(
+    urllib.error.URLError, 
+    tries=HTTP_REQUEST_RETRIES, 
+    delay=HTTP_REQUEST_DELAY_IN_SECONDS, 
+    backoff=HTTP_REQUEST_DELAY_MULTIPLIER
+)
+def url_open_with_retry(url):
+    request = urllib.request.Request(url)
+    request.add_header("Accept", "application/vnd.github.v3+json")
+    request.add_header("Authorization", "Bearer " + packagePAT)
+
+    return urllib.request.urlopen(request)
+
+# Gets dependencies of ballerina extension module from build.gradle file in module repository
+# returns: list of dependencies
+def get_dependencies(module_name):
+    try:
+        data = url_open_with_retry("https://raw.githubusercontent.com/ballerina-platform/" 
+                                    + module_name + "/master/build.gradle")
+    except:
+        print('Failed to read build.gradle file of ' + module_name)
+        sys.exit()
+
+    dependencies = []
+
+    for line in data:
+        processed_line = line.decode("utf-8")
+        if 'ballerina-platform/module' in processed_line:
+            module = processed_line.split('/')[-1]
+            if module[:-2] == module_name:
+                continue
+            dependencies.append(module[:-2])
+
+    return dependencies
+
+# Gets the version of the ballerina extension module from gradle.properties file in module repository
+# returns: current version of the module
+def get_version(module_name):
+    try:
+        data = url_open_with_retry("https://raw.githubusercontent.com/" + BALLERINA_ORG_NAME + "/"
+                                    + module_name + "/master/gradle.properties")
```

Comment:
@niveathika Can this be `main` in new repos? Just wondering whether should we parameterize to support release from other branches?

Reply:
And why is this not happing via the githuv api like eveyting else?

Reply:
@keizer619 Fixed it in 3e0a99a
@manuranga It was from legacy code, I have updated it.